devicetypes/inovelliusa/inovelli-bulb-multi-color-lzw42.src/inovelli-bulb-multi-color-lzw42.groovy

```diff
@@ -37,6 +59,539 @@
+                refreshGroup = true
+            }
+            if (refreshGroup == true) cmds << zwave.associationV2.associationGet(groupingIdentifier:i)
+            else log.debug "There are no association actions to complete for group $i"
+         }
+      } else {
+         log.debug "Association info not known for group $i. Requesting info from device."
+         cmds << zwave.associationV2.associationGet(groupingIdentifier:i).format()
+      }
+   }
+   return cmds
+}
+
+
+def setGenericTempName(temp){
+    if (!temp) return
+    def genericName
+    def value = temp.toInteger()
+    if (value <= 2000) genericName = "Sodium"
+    else if (value <= 2100) genericName = "Starlight"
+    else if (value < 2400) genericName = "Sunrise"
+    else if (value < 2800) genericName = "Incandescent"
+    else if (value < 3300) genericName = "Soft White"
+    else if (value < 3500) genericName = "Warm White"
+    else if (value < 4150) genericName = "Moonlight"
+    else if (value <= 5000) genericName = "Horizon"
+    else if (value < 5500) genericName = "Daylight"
+    else if (value < 6000) genericName = "Electronic"
+    else if (value <= 6500) genericName = "Skylight"
+    else if (value < 20000) genericName = "Polar"
+    def descriptionText = "${device.getDisplayName()} color is ${genericName}"
+    if (txtEnable) log.info "${descriptionText}"
+	sendEvent(name: "colorMode", value: "CT", descriptionText: "${device.getDisplayName()} color mode is CT")
+    sendEvent(name: "colorName", value: genericName ,descriptionText: descriptionText)
+}
+
+def setGenericName(hue){
+    def colorName
+    hue = hue.toInteger()
+    hue = (hue * 3.6)
+    switch (hue.toInteger()){
+        case 0..15: colorName = "Red"
+            break
+        case 16..45: colorName = "Orange"
+            break
+        case 46..75: colorName = "Yellow"
+            break
+        case 76..105: colorName = "Chartreuse"
+            break
+        case 106..135: colorName = "Green"
+            break
+        case 136..165: colorName = "Spring"
+            break
+        case 166..195: colorName = "Cyan"
+            break
+        case 196..225: colorName = "Azure"
+            break
+        case 226..255: colorName = "Blue"
+            break
+        case 256..285: colorName = "Violet"
+            break
+        case 286..315: colorName = "Magenta"
+            break
+        case 316..345: colorName = "Rose"
+            break
+        case 346..360: colorName = "Red"
+            break
+    }
+    def descriptionText = "${device.getDisplayName()} color is ${colorName}"
+    if (txtEnable) log.info "${descriptionText}"
+	sendEvent(name: "colorMode", value: "RGB", descriptionText: "${device.getDisplayName()} color mode is RGB")
+    sendEvent(name: "colorName", value: colorName ,descriptionText: descriptionText)
+}
+ *		Removed SmartThings related code
```

Comment:
something got borked here

Reply:
Getting an error when updating from repo in SmartThings

 Error
500: Internal Server Error
URI
/ide/device/doRepoUpdates
Reference Id
76e7a391-aaf3-4de1-a87e-92aa30b3511a
Date
Fri May 15 20:50:13 UTC 2020